### DISCUSSION TITLE
Is it possible to recover a deleted comment on a Github Issue?

### DISCUSSION THREAD

**Discussion by shuhei-fujita** [community/community#69275] [Projects and Issues] [answered]
### Select Topic Area

General

### Body

Is it possible to recover a deleted comment on a Github Issue?

I found the docs about deleted comment.
https://docs.github.com/ja/communities/moderating-comments-and-conversations/tracking-changes-in-a-comment#deleting-sensitive-information-from-a-comments-history

**Comment by AnkurBansal18** [comment]
hey @shuhei-fujita 👋,
GitHub does not provide a built-in feature for recovering or restoring deleted comments on an issue. Once a comment is deleted, it is typically considered permanent and cannot be directly recovered through GitHub's user interface.

However, there might be some potential ways to recover deleted comments:

**Ask the Commenter:** If you're the author of the deleted comment or if you know the person who deleted the comment, you can ask them to provide the content or context of the deleted comment.

**Use Version Control:** If the issue or pull request/merge request discussion was active and the comments were made during a specific period, you might be able to retrieve the deleted comments from the Git history. If the repository is public, you can view the Git history, but this may not work for private repositories.

**GitHub Enterprise:** If your organization uses GitHub Enterprise, it's possible that your administrators have more control and data recovery options. You may want to check with your organization's GitHub Enterprise administrators.

**GitHub Support:** If the deleted comment is critical or contains valuable information, you could reach out to GitHub support for assistance. They might be able to help you in specific situations, but this would likely be a last resort.

Remember that the ability to recover deleted comments may depend on the repository's settings, user roles and permissions, and the specific circumstances surrounding the deletion.

✨ Hope it helps :)

**Reply by shuhei-fujita** [reply]
@AnkurBansal18 
Thank you!

I found the docs about deleted comment.
https://docs.github.com/ja/communities/moderating-comments-and-conversations/tracking-changes-in-a-comment#deleting-sensitive-information-from-a-comments-history

> However, there might be some potential ways to recover deleted comments:

I think so too.
I will find a way.

**Reply by lukaseder** [reply]
Just a hint, there's been a series of spammers who comment on github issues and immediately delete the comment again (probably automatically). The goal is to have github send notification links to malware. I would like to report these spammers, but with the comments deleted, I obviously cannot.

**Reply by polarathene** [reply]
I'd appreciate it if the comment author could at least retain the right to their comment, with the authority to delete it from GH records permanently rather than whomever else invokes the deletion of the comment.

It's incredibly frustrating when I spend a fair amount of time providing helpful details and references in a comment and that unintentionally triggers a project owner for example to delete it (_a recent example was I chimed in on a project issue that had been stalled for over 6 months, provided some suggestions about going forward and asking about the current blocker. In that same comment I did reference some alternative projects where the issue was not present which I assume triggered the project owner_).

Sometimes my comments can be information dense and formatted in a way that looks like it's from an AI / LLM output, but I've been doing that for over a decade, despite the information being of value, all it takes is someone (or an automated tool) to gloss over it and for whatever reason delete it...

If I reference the comment in my own personal notes or via other projects/communities, I've learned that anything quite important that I submit online can at a whim be removed, so I've become a tad paranoid with trusting my comments on platforms not to be prone to loss... requiring backup copies to be archived :\

I'd be at much more ease if I could at least retrieve my comments when someone else actioned a delete.

**Comment by queenofcorgis** [comment, marked as answer]
Thanks for posting in the GitHub Community, @shuhei-fujita ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. I've gone ahead and moved it for you. Good luck!

**Reply by shuhei-fujita** [reply]
Thank you for your answer and for moving the post to the GitHub product Discussions, @queenofcorgis!
